addons/event_sale/static/src/js/sale_product_field.js

```diff
@@ -56,3 +60,37 @@ patch(SaleOrderLineProductField.prototype, {
         );
     },
 });
+
+export class EventRegistrationQtyField extends FloatField {
+    setup() {
+        super.setup();
+        this.action = useService("action");
+        this.uiService = useService("ui");
+        this.state = useState({ value: this.props.record.data.product_uom_qty });
+        useEffect(() => {
+            if ((this.state.value !== this.props.record.data.product_uom_qty || this.props.record.isNew) && this.props.record.data.state === 'sale' && this.props.record.data.event_id) {
+                this.state.value = this.props.record.data.product_uom_qty;
+                this.update();
+            }
+        });
+    }
+    async update() {
+        this.uiService.block();
+        await this.props.record.model.root.save({ stayInEdition: true });
+        const orderLineIds = this.env.model.root.data.order_line.resIds;
+        try {
+            const action = await this.env.model.orm.call("sale.order.line", "update_registrations_qty", [orderLineIds]);
+            this.action.doAction(action);
+        } finally {
+            this.uiService.unblock();
+        }
+        return true;
+    }
+}
+
+export const eventRegistrationQtyField = {
+    ...floatField,
+    component: EventRegistrationQtyField,
+};
+
+registry.category("fields").add("sol_product_quantity", eventRegistrationQtyField);
```

Comment:
I don't think that adding a specific widget for the quantity field is the way to go.

It will prevent modularity if we want to do the same thing for event booth for example.
(And globally I don't think that "event_sale" should be the module to define the quantity widget on a SO line, not really its job / responsibility).

Ideally this logic should be inside, or at least be coupled with what is done in "SaleOrderLineProductField" (with "_onProductUpdate" etc).
Which may prove very difficult as the product field does not receive quantity information (as it's a separate field).
Meaning we'd have to somehow rework the whole thing...

Note that the spec says:
> Disclaimer: Those are apparently tricky parts of Odoo's code. If it is very trick and/or requires a ton of specific code, let me know and we may cancel.

Not sure if worth it really, but tell me what you think 😄 
(Can also be discussed at GR2 of course).